mlflow/projects/__init__.py

```diff
@@ -176,13 +177,20 @@ def _get_storage_dir(storage_dir):
 
 
 def _expand_uri(uri):
-    if _GIT_URI_REGEX.match(uri):
+    if _is_git_uri(uri):
         return uri
     return os.path.abspath(uri)
 
 
-def _is_local_path(uri):
-    return urllib.parse.urlparse(uri).scheme == ''
+def _is_git_uri(uri):
+    return _GIT_URI_REGEX.match(uri) is not None
+
+
+def _should_use_temp_dir(parsed_uri):
+    """
+    Returns True if we should fetch the project at the passed-in URI into a temporary directory.
+    """
+    return _is_git_uri(parsed_uri) or mlflow.data.is_uri(parsed_uri)
```

Comment:
I don't think it's worth relying on an external implementation for this:

```
    from six.moves import urllib
    parsed_uri = urllib.parse.urlparse(string)
    return len(parsed_uri.scheme) > 0
```

Much harder to understand what it's doing.